indico/modules/events/registration/__init__.py

```diff
@@ -103,6 +106,20 @@ def _get_management_roles(sender, **kwargs):
     return RegistrationRole
 
 
+@signals.get_placeholders.connect_via('registration-email')
+def _get_registration_placeholders(sender, registration, **kwargs):
+    from indico.modules.events.registration.placeholders.registrations import (FirstNamePlaceholder, LastNamePlaceholder,
+                                                                               EventTitlePlaceholder, EventURLPlaceholder,
+                                                                               IDPlaceholder, LinkPlaceholder)
+
+    yield FirstNamePlaceholder
+    yield LastNamePlaceholder
+    yield EventTitlePlaceholder
+    yield EventURLPlaceholder
```

Comment:
I posted some comments on an older commit related to this (you probably have them in your emails)